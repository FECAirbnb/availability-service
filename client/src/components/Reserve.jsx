/* eslint-disable react/jsx-closing-tag-location */
/* eslint-disable react/self-closing-comp */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react/prop-types */
/* eslint-disable no-unused-vars */
/* eslint-disable import/extensions */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/button-has-type */
/* eslint-disable react/jsx-one-expression-per-line */
import React from 'react';
import Calendar from './Calendar.jsx';
// import AddGuests from './AddGuests.jsx';

class Reserve extends React.Component {
  constructor(props) {
    super(props);
    this.container = React.createRef();
    this.state = {
      open: false,
      clickGuest: false,
      checkIn: 'Check in',
      checkOut: 'Check out',
      guestCount: 1
    };

    this.handleDropdownClick = this.handleDropdownClick.bind(this);
    this.handleOutsideClick = this.handleOutsideClick.bind(this);
    this.handleDateChange = this.handleDateChange.bind(this);
    this.handleStatechange = this.handleStatechange.bind(this);
    this.handleGuestCount = this.handleGuestCount.bind(this);
    this.handleGuestDropdown = this.handleGuestDropdown.bind(this);
  }

  componentDidMount() {
    document.addEventListener('mousedown', this.handleOutsideClick);
  }

  componentWillUnmount() {
    document.removeEventListener('mousedown', this.handleOutsideClick);
  }

  handleDropdownClick() {
    this.setState(state => {
      return {
        open: true
      };
    });
  }

  handleGuestDropdown() {
    if (!this.state.clickGuest) {
      this.setState(state => {
        return {
          clickGuest: true
        };
      });
    } else {
      this.setState(state => {
        return {
          clickGuest: false
        };
      });
    }
  }

  handleOutsideClick(e) {
    if (this.container.current && !this.container.current.contains(e.target)) {
      this.setState({
        open: false,
        clickGuest: false
      });
    }
  }

  handleDateChange(date) {
    if (this.state.checkIn === 'Check in') {
      this.setState({
        checkIn: date
      });
    } else {
      this.setState({
        checkOut: date
      });
    }
  }

  handleStatechange() {
    this.setState({
      open: false,
      checkIn: 'Check in',
      checkOut: 'Check out'
    });
    this.props.onClick();
  }

  handleGuestCount() {
    let guestStr = '';
    if (this.state.guestCount === 1) {
      guestStr = 'guest';
    } else {
      guestStr = 'guests';
    }
    return guestStr;
  }

  handleArrowChange() {
    let arrow = '';
    if (this.state.clickGuest === false) {
      arrow = (
        <img
          src="https://image.flaticon.com/icons/png/128/566/566015.png"
          data-src="https://image.flaticon.com/icons/png/128/566/566015.png"
          srcSet="https://image.flaticon.com/icons/svg/566/566015.svg 4x"
          alt="Down chevron free icon"
          title="Down chevron free icon"
          width="64"
          height="64"
          className="arrow"
        ></img>
      );
    } else {
      arrow = (
        <img
          src="https://image.flaticon.com/icons/png/128/566/566014.png"
          data-src="https://image.flaticon.com/icons/png/128/566/566014.png"
          srcSet="https://image.flaticon.com/icons/svg/566/566014.svg 4x"
          alt="Up chevron free icon"
          title="Up chevron free icon"
          width="64"
          height="64"
          className="arrow"
        ></img>
      );
    }
    return arrow;
  }

  render() {
    return (
      <div className="reserve" ref={this.container}>
        <div>
          <span className="price">${this.props.state.data[0].price}</span>
          <span className="per-night"> per night</span>
        </div>
        <div>
          <span role="img" aria-label="Rating 1 out of 5" className="_m5wjo84">
            4.99
          </span>
          <span className="per-night">(6 reviews)</span>
        </div>
        <div className="date-picker">
          <div id="check-in" onClick={this.handleDropdownClick}>
            {this.state.checkIn}
          </div>

          <div id="check-out" onClick={this.handleDropdownClick}>
            {this.state.checkOut}
          </div>
        </div>
        <div>
          {this.state.open && (
            <div className="dropdown">
              <Calendar state={this.props.state} handleDateChange={this.handleDateChange} />
            </div>
          )}
        </div>
        <div className="add-guest">
          <div id="guest-count" onClick={this.handleGuestDropdown}>
            {this.state.guestCount} {this.handleGuestCount()}
          </div>

          <div id="arrow" onClick={this.handleGuestDropdown}>
            {this.handleArrowChange()}
          </div>
        </div>
        <div>
          {this.state.clickGuest && (
            <div className="dropdown">
              <div>
                <div>hello</div>
              </div>
            </div>
          )}
        </div>
        <div>
          <button id="reserve-btn" type="submit" onClick={this.handleStatechange}>
            RESERVE
          </button>
        </div>
      </div>
    );
  }
}

export default Reserve;
