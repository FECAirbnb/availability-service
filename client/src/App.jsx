/* eslint-disable react/no-unused-state */
/* eslint-disable no-console */
/* eslint-disable class-methods-use-this */
/* eslint-disable import/extensions */
/* eslint-disable react/destructuring-assignment */
import React from 'react';
import axios from 'axios';
import Reserve from './components/Reserve.jsx';
import styles from './style.module.css';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: null,
      locationId: 1
    };
    this.bookDates = this.bookDates.bind(this);
    this.updateState = this.updateState.bind(this);
  }

  componentDidMount() {
    axios
      .get(`/api/reserve/${this.state.locationId}`)
      .then(result => {
        this.setState({
          data: result.data
        });
      })
      .catch(err => {
        throw err;
      });
  }

  updateState() {
    axios
      .get(`/api/reserve/${this.state.locationId}`)
      .then(result => {
        this.setState({
          data: result.data
        });
      })
      .catch(err => {
        throw err;
      });
  }

  bookDates() {
    const checkIn = document.getElementById('check-in').textContent;
    const checkOut = document.getElementById('check-out').innerText;
    let checkInDate = new Date(checkIn);
    let checkOutDate = new Date(checkOut);

    checkInDate = checkInDate.toISOString();
    checkOutDate = checkOutDate.toISOString();
    axios
      .get(`/api/reserve/dates/:${checkInDate}:${checkOutDate}`)
      .then(result => {
        return result.data;
      })
      .then(dates => {
        const { locationId } = this.state;
        axios.post(`/api/reserve/book/${this.state.locationId}`, {
          dates,
          locationId
        });
      })
      .catch(err => {
        throw err;
      });
    this.updateState();
  }

  renderView() {
    if (this.state.data === null) {
      return <div> No locations </div>;
    }
    return (
      <div>
        <Reserve onClick={this.bookDates} state={this.state} />
      </div>
    );
  }

  render() {
    return <div className={styles.main}>{this.renderView()}</div>;
  }
}

export default App;
