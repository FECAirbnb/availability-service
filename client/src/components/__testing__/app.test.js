/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from 'react';
import toJson from 'enzyme-to-json';
import App from '../../App';
import Reserve from '../Reserve';
import { shallow, mount } from '../../enzyme';

const sampleData = [
  {
    id: 1,
    name: 'Bora bora',
    price: 1450,
    cleaning_fee: 100,
    service_fee: 100,
    date: '2020-03-01T07:00:00.000Z',
    Location_id: 1,
    Dates_id: 1
  }
];

describe('App component', () => {
  it('renders correctly enzyme', () => {
    const wrapper = shallow(<App />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  const wrapper = mount(<App />);

  beforeEach(() => {
    wrapper.setState({
      data: [
        {
          id: 1,
          name: 'Bora bora',
          price: 1450,
          cleaning_fee: 100,
          service_fee: 100,
          date: '2020-03-01T07:00:00.000Z',
          Location_id: 1,
          Dates_id: 1
        }
      ]
    });
    wrapper.setState({ locationId: 1 });
  });

  it('should have a location number as a state', () => {
    expect(wrapper.state('locationId')).toEqual(1);
  });
});
