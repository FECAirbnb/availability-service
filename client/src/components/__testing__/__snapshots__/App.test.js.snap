// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly enzyme 1`] = `
<div
  className="main"
>
  <div>
     No locations 
  </div>
</div>
`;
